### PR TITLE
Implemented proceses priority

### PR DESCRIPTION
**Pull request by AndreaCatania**
With this PR I've added a way to customize the process execution order of nodes.

### CODE REVIEW

File: scene/main/node.cpp [hunk 1]
```diff
@@ -1388,6 +1396,11 @@ bool Node::is_greater_than(const Node *p_node) const {
 	return res;
 }
 
+bool Node::is_priority_than(const Node *p_node) const {
```
Comment (karroffel) [inline scene/main/node.cpp]: `is_priority_more_tasty_than`? :P I'd suggest `is_priority_higher_than` or something similar, the current name is confusing :sweat_smile:

Comment (AndreaCatania) [reply scene/main/node.cpp]: Yes better :D

**Comment by KezzB** [comment]
Just wanted to chime in and say thanks for this PR!

This is perfect for what I'm working on right now, which is a game framework where I'm aiming to put together some common game functionality as ready-to-use nodes. Things like directional movement, mouse look for weapons, weapon archetypes, projectiles, responses to collisions etc.

With a bunch of self contained nodes, this will help me to make sure all their functionality resolves in the right order.

Hope to see this in a future update, looking forward to using it!

File: scene/main/node.cpp [hunk 2]
```diff
@@ -809,6 +809,14 @@ bool Node::is_processing_internal() const {
 	return data.idle_process_internal;
 }
 
+void Node::set_process_priority(int p_priority) {
```
Comment (reduz) [inline scene/main/node.cpp]: I would check that the node is in that group before forcing it to be changes, you can check with is_processing, is physics_processing, etc. and so on.

**Review by reduz — CHANGES_REQUESTED** [review]

**Pull request merged**